glean-core/src/lib.rs

```diff
@@ -213,24 +214,35 @@ impl Glean {
             let _scanning_thread = upload_manager.scan_pending_pings_directories();
         }
 
-        Ok(Self {
+        let (start_time, start_time_is_corrected) = local_now_with_offset();
+        let this = Self {
             upload_enabled: cfg.upload_enabled,
             // In the subprocess, we want to avoid accessing the database entirely.
             // The easiest way to ensure that is to just not initialize it.
             data_store: None,
             event_data_store,
             core_metrics: CoreMetrics::new(),
+            additional_metrics: AdditionalMetrics::new(),
             database_metrics: DatabaseMetrics::new(),
             internal_pings: InternalPings::new(),
             upload_manager,
             data_path: PathBuf::from(&cfg.data_path),
             application_id,
             ping_registry: HashMap::new(),
-            start_time: local_now_with_offset(),
+            start_time,
             max_events: cfg.max_events.unwrap_or(DEFAULT_MAX_EVENTS),
             is_first_run: false,
             debug: DebugOptions::new(),
-        })
+        };
+
+        // Can't use `into_inner_record` above, because we needed a valid `Glean` first.
```

Comment:
What is `into_inner_record`?

Reply:
A thing that doesn't exist and should have never existed (I'm glad I didn't push the code, it was like 50 lines more for no good reason).
I'll change the comment.